locator/tablets.cc

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "locator/tablet_replication_strategy.hh"
+#include "locator/tablets.hh"
+#include "types/types.hh"
+#include "types/tuple.hh"
+#include "types/set.hh"
+#include "utils/hash.hh"
+#include "db/system_keyspace.hh"
+#include "cql3/query_processor.hh"
+#include "cql3/untyped_result_set.hh"
+#include "replica/database.hh"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/coroutine/maybe_yield.hh>
+
+namespace locator {
+
+seastar::logger tablet_logger("tablets");
+
+std::ostream& operator<<(std::ostream& out, const tablet_replica& r) {
+    return out << r.host << ":" << r.shard;
+}
+
+const tablet_map& tablet_metadata::get_tablet_map(table_id id) const {
+    try {
+        return _tablets.at(id);
+    } catch (const std::out_of_range&) {
+        throw std::runtime_error(format("Tablet map not found for table {}", id));
+    }
+}
+
+tablet_map& tablet_metadata::get_tablet_map(table_id id) {
+    return const_cast<tablet_map&>(
+        const_cast<const tablet_metadata*>(this)->get_tablet_map(id));
+}
+
+void tablet_metadata::set_tablet_map(table_id id, tablet_map map) {
+    _tablets[id] = std::move(map);
+}
+
+future<> tablet_metadata::clear_gently() {
```

Comment:
how about `utils::clear_gently(_tablets)`?

Reply:
and also, this should be noexcept.

Reply:
Ok.